packages/xo-server/src/xapi/mixins/patching.mjs

```diff
@@ -587,16 +603,117 @@ const methods = {
     throw new Error('non pool-wide install not implemented')
   },
 
-  async rollingPoolUpdate($defer, parentTask, { xsCredentials } = {}) {
+  async xsCdnUpdate(hosts, hash) {
+    if (hosts === undefined) {
+      hosts = Object.values(this.objects.indexes.type.host)
+    }
+
+    if (hash === undefined) {
+      hash = (await this._fetchXsUpdatesEndpoint(hosts[0])).hash
+    }
+
+    // Hosts need to be updated one at a time starting with the pool master
+    hosts = hosts.sort(({ $ref }) => ($ref === this.pool.master ? -1 : 1))
+
+    for (const host of hosts) {
+      await this.callAsync('host.apply_updates', host.$ref, hash)
+    }
+
+    await this._fetchXsUpdatesEndpoint(REMOVE_CACHE_ENTRY, hosts[0])
```

Comment:
I think you can use debounceWithKey with a delay of 0, which will effectively remove the method from the cahce as soon as it's done. By default it is also removed from debounce cache on failure

Reply:
So something like that?
```js
...
await debouncewithKey(this._fetchXsUpdatesEndpoint, 0)(hosts[0])
```

I find it weird and not explicit

Reply:
I was under the assumption that  we only wanted to keep the cache during an in flight query, and not LISTING_DEBOUNCE_TIME_MS